src/numericalAnalysis.cpp

```diff
@@ -0,0 +1,119 @@
+#include "mantella_bits/numericalAnalysis.hpp"
+
+// C++ standard library
+#include <cmath>
+#include <limits>
+
+// Mantella
+#include "mantella_bits/assert.hpp"
+
+namespace mant {
+  double brent(
+      std::function<double(double)> objectiveFunction,
+      double lowerBound,
+      double upperBound,
+      double xTolerance) {
+    double lowerBoundObjectiveValue = objectiveFunction(lowerBound);
+    double upperBoundObjectiveValue = objectiveFunction(upperBound);
+
+    verify(lowerBoundObjectiveValue * upperBoundObjectiveValue <= 0.0, "brent: The lower bound and upper bound objective value must have a different sign.");
+
+    double previousUpperBound = lowerBound; // only on initialisation: previousUpperBound = lowerBound
+    double previousUpperBoundObjectiveValue = lowerBoundObjectiveValue;
+    double boundDistance = upperBound - lowerBound;
+    double toleranceCriteria = boundDistance;
+
+    double middleOfBounds;
+    double boundObjectiveValueRatio;
+    double interpolationNumerator;
+    double interpolationFactor;
+    double boundPreviousBoundRatio;
+
+    int iter = 0;
```

Comment:
First: This should be marked as `unsigned`
Second: We decided to always stick to `arma::uword` is we need an unsigned integer. (The only exception is MPI-based code, as MPI is very picky about requiring signed integers, even if MPI will later raise an exception if the value will ever become negative...)